torch/export/dynamic_shapes.py

```diff
@@ -805,8 +805,13 @@ def _check_dynamic_shapes(
     using combined args + kwargs as reference for inputs structure.
     """
     from torch._dynamo.exc import UserError, UserErrorType
+    from torch._export.non_strict_utils import _flatten_dynamic_shapes
 
-    if dynamic_shapes is None or len(dynamic_shapes) == 0:
+    if (
+        dynamic_shapes is True
```

Comment:
You sure you need the `True` here? The `None` value plays a couple of roles here, one for subtrees (including tensors as a whole) and one for dimensions. I believe you want `True` to apply only to dimensions, like `Dim`.

Reply:
Yeah, this is in case of `dynamic_shapes=True` - we'll let everything be dynamic